File: 2.Implement/Delivery_Server/src/main/java/com/delivery/app/Delivery/data/entity/Stock.java
package com.delivery.app.Delivery.data.entity;

import com.sun.istack.NotNull;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity @Table(name="stocks")
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Stock {
    @Id @NotNull
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer price;

    @Column(nullable = false)
    private LocalDateTime releasedDate;

    @Column(nullable = false)
    private String manufacturer;

    @Column(nullable = false)
    private Integer weight;
}

File: 2.Implement/DeliveryApp/app/src/main/java/com/project/deliveryapp/data/Stock.kt
package com.project.deliveryapp.data

import android.graphics.Bitmap
import android.media.Image
import java.sql.Date
import java.time.LocalDateTime


data class Stock(
    val imageBitmap: Bitmap?,
    val id: Long,
    val name: String,
    val releaseDate: LocalDateTime,
    val manufacturer: String,
    val weight: Float,
    val price: Long
)
